include/ideep/operators/inner_product.hpp

```diff
@@ -195,30 +195,41 @@ struct inner_product_forward : public dnnl::inner_product_forward {
        : primitive_desc({aprop_kind, src_desc, weights_desc, dst_desc},
                         op_attr, aengine);
 
+    // reorder src, weight, dst if needed
     auto expected_src = src.reorder_if_differ_in(pd.src_desc(), src_attr);
     auto expected_weights = weights.reorder_if_differ_in(pd.weights_desc(), weights_attr);
-    dst.reinit_if_possible(pd.dst_desc());
-    if (!dst_scales.empty() && dst.get_data_type() != data_type::f32) {
-      dst.set_scale(dst_scales_in);
+
+    // [ Note output buffer]
+    // In this case, dst is an empty ideep tensor, can be re-init
+    // If dst is not empty, ideep must wirte result to dst's memory and it is caller's duty to 
+    // make sure dst is big enough to hold the result 
```

Comment:
Why is the size of the dst buffer a concern? If the format of the provided dst doesn't match what oneDNN wants, we would create a new buffer `expected_dst` and later reorder the content to `dst`, right?

Reply:
Because the buffer is allocated by aten, and shared toideep.
We cannot allocate buffer in ideep and share back to aten without copy since ideep tensor is a temporary variable and the buffer will be released after ideep tensor died.